SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -32,6 +32,37 @@ extension Array {
     }
 }
 
+extension Array {
+    /// Separates this array's elements into chunks with a given length.
+    ///
+    /// If `length` is not a positive number or if it's greater than or equal to this array's length, we return a new array
+    /// whose sole element is this array itself.
+    ///
+    /// - parameter length: Maximum number of elements in each chunk.
+    /// - parameter includeRemainder: If set to `true`, include the last chunk in returned array even if its length is less than
+    ///   `length`. If set to `false`, discard the last chunk if its length is less than `length`. Defaults to `true`.
+    ///
+    /// - returns: New array of chunks of elements in this array.
+    public func ip_chunks(ofLength length: Int, includeRemainder: Bool = true) -> [[Element]] {
+        guard length > 0 else { return [self] }
+
+        var mutable = self
+        var chunks = [[Element]]()
+
+        while mutable.count >= length {
+            let next = Array(mutable.prefix(length))
+            chunks.append(next)
+            mutable.removeFirst(length)
```

Comment:
I'm a little concerned about the performance characteristics of this implementation.  (I wouldn't necessarily be concerned for a known use case in a particular product, but since we're proposing a general purpose library function, I want to ensure that it's performant.)

I assume that `removeFirst` requires an allocation and an array copy for each call?  If we expect to be using this for large arrays (or for array / `length` combinations that will hit this line many times), would we be better off to use another algorithm that peels elements off the *end* of the array?  Perhaps even better, avoid making `remove*` calls altogether and track the  the region to copy "by hand," avoiding the additional allocations and copies that `removeFirst` presumably introduces?

(BTW, this is exactly why I'm worried about ensuring order, or not, in my "overall" comment in this PR.  Because, for example, we *could* ship this implementation, which guarantees order, and then decide to move to a "reverse" implementation that reverses the order of the blocks that are peeled off.  It would be best to decide on, and *document,* a strategy once.)